### PR TITLE
tree-wide: fix malformed meta.maintainers

### PR DESCRIPTION
**Pull request by tilpner**
###### Motivation for this change

These packages stood out while processing packages.json.gz,
where the maintainers field of some packages wasn't the expected list of sets,
but a nested list, or a string.

###### Notify maintainers

cc @sterfield @1000101

### CODE REVIEW

File: pkgs/development/libraries/at-spi2-core/default.nix [hunk 1]
```diff
@@ -45,7 +45,7 @@ stdenv.mkDerivation rec {
     description = "Assistive Technology Service Provider Interface protocol definitions and daemon for D-Bus";
     homepage = https://gitlab.gnome.org/GNOME/at-spi2-core;
     license = licenses.lgpl2Plus; # NOTE: 2018-06-06: Please check the license when upstream sorts-out licensing: https://gitlab.gnome.org/GNOME/at-spi2-core/issues/2
-    maintainers = with maintainers; [ jtojnar gnome3.maintainers ];
+    maintainers = with maintainers; [ jtojnar ] ++ gnome3.maintainers;
```
Comment (worldofpeace) [inline pkgs/development/libraries/at-spi2-core/default.nix]: jtojnar is already a member of the gnome3 maintainers list

File: pkgs/development/libraries/at-spi2-atk/default.nix [hunk 2]
```diff
@@ -42,7 +42,7 @@ stdenv.mkDerivation rec {
     description = "D-Bus bridge for Assistive Technology Service Provider Interface (AT-SPI) and Accessibility Toolkit (ATK)";
     homepage = https://gitlab.gnome.org/GNOME/at-spi2-atk;
     license = licenses.lgpl2Plus; # NOTE: 2018-06-06: Please check the license when upstream sorts-out licensing: https://gitlab.gnome.org/GNOME/at-spi2-atk/issues/2
-    maintainers = with maintainers; [ jtojnar gnome3.maintainers ];
+    maintainers = with maintainers; [ jtojnar ] ++ gnome3.maintainers;
```
Comment (worldofpeace) [inline pkgs/development/libraries/at-spi2-atk/default.nix]: jtojnar is already a member of the gnome3 maintainers list

Comment (worldofpeace) [inline pkgs/development/libraries/at-spi2-atk/default.nix]:
```suggestion
    maintainers = gnome3.maintainers;
```

File: pkgs/development/libraries/at-spi2-core/default.nix (hunk 1 above)
Comment (worldofpeace) [inline pkgs/development/libraries/at-spi2-core/default.nix]:
```suggestion
    maintainers = gnome3.maintainers;
```

File: pkgs/development/libraries/at-spi2-atk/default.nix (hunk 2 above)
Comment (tilpner) [reply pkgs/development/libraries/at-spi2-atk/default.nix]: Do you want me to remove them?
It's possible that @jtojnar leaves the gnome3 maintainer list without intending to give up maintenance of at-spi2-atk.

Comment (worldofpeace) [reply pkgs/development/libraries/at-spi2-atk/default.nix]: :rofl: I'd think jtojnar would have given up on NixOS if it reached that point.

Nothing strange should happen here if they've in this list twice?

**Comment by tilpner** [comment]
@worldofpeace I agree, it's now using `maintainers = gnome3.maintainers;`

**Comment by worldofpeace** [comment]
Sounds good @tilpner, no further actions needed?

**Comment by tilpner** [comment]
We could give @sterfield another day to speak up, I'm not sure they want to be maintainer. But otherwise no, merge whenever. :)

File: pkgs/development/python-modules/trezor/default.nix [hunk 3]
```diff
@@ -44,6 +44,6 @@ buildPythonPackage rec {
     description = "Python library for communicating with TREZOR Bitcoin Hardware Wallet";
     homepage = "https://github.com/trezor/trezor-firmware/tree/master/python";
     license = lib.licenses.gpl3;
-    maintainers = with lib.maintainers; [ np prusnak mmahut "1000101" ];
+    maintainers = with lib.maintainers; [ np prusnak mmahut ] ++ [ lib.maintainers."1000101" ];
   };
```
Comment (jtojnar) [inline pkgs/development/python-modules/trezor/default.nix]: Perhaps use `meta = with lib;`.

Comment (jtojnar) [reply pkgs/development/python-modules/trezor/default.nix]: Also you can use `lib.maintainers.foo`  inside the first list – `with` statement just extends scope with unbound names.

Comment (jtojnar) [reply pkgs/development/python-modules/trezor/default.nix]: Also you can use `lib.maintainers.foo`  inside the first list – `with` statement just extends scope with unbound names.

Comment (tilpner) [reply pkgs/development/python-modules/trezor/default.nix]: Very good point, one moment

File: pkgs/development/python-modules/shamir-mnemonic/default.nix [hunk 4]
```diff
@@ -17,6 +17,6 @@ buildPythonPackage rec {
     description = "Reference implementation of SLIP-0039";
     homepage = "https://github.com/trezor/python-shamir-mnemonic";
     license = lib.licenses.mit;
```
Comment (1000101) [inline pkgs/development/python-modules/shamir-mnemonic/default.nix]: You might also want to change `meta` to `meta = with lib;` (as you did with pkgs/development/python-modules/trezor/default.nix) to keep it consistent. I'd be really grateful for this change!

**Comment by 1000101** [comment]
@tilpner Just out of curiosity - how precisely did you find this? Many thanks for cleaning up my mess!!!

**Comment by sterfield** [comment]
Hey all,

I have created this package quite some time ago, but unfortunately, I don't have a lot of time to spend on NixOS (plus I'm afraid to say that my knowledge on NixOs stayed quite low).

But IIRC, this package is fairly simple, so I could still try to maintain it from time to time.

Tell me if you want to remove me from the list or keep me. Either way, I'm fine.

Thanks,

**Comment by tilpner** [comment]
@1000101 I was processing packages.json to sort maintainers by the count of their packages, and there were `meta.maintainers` lists that contained lists or contained just plain strings, instead of being the expected lists of attrsets.

As you know, Nix handles identifiers starting with digits differently than those starting with letters. To prevent this error in the future, one option would be to rename your attribute. It's fine to keep it like this, you just have double-check every time that you put an attrset instead of a string into the maintainers field.

@sterfield In that case, I'll keep you as maintainer, because you will eventually receive notifications when changes are proposed to sickrage. If you are not any longer a user of nixpkgs and sickrage in particular, feel free to remove yourself as a maintainer.

**Review by worldofpeace — APPROVED** [review]

**Pull request merged**